content/docs/examples/advanced-gateways/egress-gateway-tls-origination/index.md

```diff
@@ -1,22 +1,284 @@
-Finally, you direct traffic from application pods inside the mesh to the NGINX server outside the mesh through
-the egress gateway.
+## Before you begin
 
-## Generate client and server certificates and keys
+*   Setup Istio by following the instructions in the [Installation guide](/docs/setup/).
+
+*   Start the [sleep]({{< github_tree >}}/samples/sleep) sample
+    which will be used as a test source for external calls.
+
+    If you have enabled [automatic sidecar injection](/docs/setup/kubernetes/sidecar-injection/#automatic-sidecar-injection), do
+
+    {{< text bash >}}
+    $ kubectl apply -f @samples/sleep/sleep.yaml@
+    {{< /text >}}
+
+    otherwise, you have to manually inject the sidecar before deploying the `sleep` application:
+
+    {{< text bash >}}
+    $ kubectl apply -f <(istioctl kube-inject -f @samples/sleep/sleep.yaml@)
+    {{< /text >}}
+
+    Note that any pod that you can `exec` and `curl` from would do.
+
+*   Create a shell variable to hold the name of the source pod for sending requests to external services.
+    If you used the [sleep]({{<github_tree>}}/samples/sleep) sample, run:
+
+    {{< text bash >}}
+    $ export SOURCE_POD=$(kubectl get pod -l app=sleep -o jsonpath={.items..metadata.name})
+    {{< /text >}}
+
+## Perform TLS origination with an egress gateway
+
+This section describes how to perform the same TLS origination as in the
+[TLS Origination for Egress Traffic](/docs/examples/advanced-gateways/egress-tls-origination/) example,
+only this time using an egress gateway. Note that in this case the TLS origination will
+be done by the egress Gateway server, as opposed to by the sidecar in the previous example.
+
+1.  Define a `ServiceEntry` for `edition.cnn.com`:
+
+    {{< text bash >}}
+    $ kubectl apply -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: ServiceEntry
+    metadata:
+      name: cnn
+    spec:
+      hosts:
+      - edition.cnn.com
+      ports:
+      - number: 80
+        name: http-port
+        protocol: HTTP
+      - number: 443
+        name: http-port-for-tls-origination
+        protocol: HTTP
+      resolution: DNS
+    EOF
+    {{< /text >}}
+
+1.  Verify that your `ServiceEntry` was applied correctly by sending a request to [http://edition.cnn.com/politics](https://edition.cnn.com/politics).
+
+    {{< text bash >}}
+    $ kubectl exec -it $SOURCE_POD -c sleep -- curl -sL -o /dev/null -D - http://edition.cnn.com/politics
+    HTTP/1.1 301 Moved Permanently
+    ...
+    location: https://edition.cnn.com/politics
+    ...
+
+    command terminated with exit code 35
+    {{< /text >}}
+
+    Your `ServiceEntry` was configured correctly if you see _301 Moved Permanently_ in the output.
+
+1.  Create an egress `Gateway` for _edition.cnn.com_, port 443, and a destination rule for
+    sidecar requests that will be directed to the egress gateway.
+
+    Choose the instructions corresponding to whether or not you have
```

Comment:
This became irrelevant lately. mTLS could be applied regardless if Istio was deployed with or without mTLS. So it is up to the user to decide if they want mTLS or not.

Reply:
Yes, but I was planning to make functional changes to the examples in a followup PR. This one is just trying to reorg and clarify the material without changing the actual steps.

Reply:
Good approach, agreed.